src/Knp/Component/Pager/Exception/PageLimitInvalidException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Knp\Component\Pager\Exception;
+
+use OutOfRangeException;
+use Throwable;
+
+final class PageLimitInvalidException extends OutOfRangeException
+{
+    private int $limit;
+
+    public function __construct(?string $message, int $limit, ?Throwable $previousException = null)
```

Comment:
I would avoid overriding original constructor, because you lose the code argument. It would be better to use a static property for $limit and a static constructor.